docs/components/index.md

```diff
@@ -0,0 +1,315 @@
+
+Parent component:
+
+```cshtml
+<ChildComponent bind-Year="@ParentYear" />
+
+@functions {
+    public int ParentYear { get; set; } = 1979;
+}
+```
+
+Child component:
+
+```cshtml
+<div> ... </div>
+
+@functions {
+    public int Year { get; set; }
+    public Action<int> ValueChanged { get; set; }
+}
+```
+
+**User-defined binding**
+
+Other mappings for `bind` can be created:
+
+```cshtml
+[BindElement("ul", "foo", "myvalue", "myevent")]
+public class BindAttributes
+{
+}
+```
+
+*MyComponent.cshtml*:
+
+```cshtml
+<ul bind-foo="@SomeExpression" />
+```
+
+**Bind value and change handler attributes**
+
+`bind` can be specified with any value attribute name and change handler attribute name. The following example binds the current value of `SomeExpression` to `myvalue` and a change handler lambda to `myevent`:
+
+```cshtml
+<ul bind-myvalue-myevent="@SomeExpression">
+    ...
+</ul>
+```
+
+## Event handling
+
+Blazor offers two event handling features, `@onclick` and `@onchange`. The following code calls the `UpdateHeading` method when the button is selected in the UI:
+
+```cshtml
+<button class="btn btn-primary" @onclick(UpdateHeading)>
+    Update heading
+</button>
+```
+
+The following code calls the `CheckboxChanged` method when the checkbox is changed in the UI:
+
+```cshtml
+<input type="checkbox" class="form-check-input" 
+    id="callsMethodCheck" @onchange(CheckboxChanged)>
+```
+
+`OnInit` and `OnInitAsync` execute code after the component has been initialized:
+
+```csharp
+protected override void OnInit()
+{
+    ...
+}
+```
+
+```csharp
+protected override async Task OnInitAsync()
+{
+    ...
+}
+```
+
+`OnParametersSet` and `OnParametersSetAsync` are called when a component has received parameters from its parent and the values are assigned to properties. These methods are executed after `OnInit` during component initialization.
+
+```csharp
+protected override void OnParametersSet()
+{
+    ...
+}
+```
+
+```csharp
+protected override async Task OnParametersSetAsync()
+{
+    ...
+}
+```
+
+`SetParameters` can be overridden to execute code before parameters are set:
+
+```csharp
+public override void SetParameters(ParameterCollection parameters)
+{
+    ...
+    
+    base.SetParameters(parameters);
+}
+```
+
+`ShouldRender` can be overridden to suppress refreshing of the UI. If the implementation returns `true`, the UI is refreshed. Even if `ShouldRender` is overridden, the component is always initially rendered.
+
+```csharp
+protected override bool ShouldRender()
+{
+    var renderUI = true;
+
+    return renderUI;
+}
+```
+
+## Component disposal with IDisposable
+
+If a component implements [IDisposable](/dotnet/api/system.idisposable), the router disposes the component when the user navigates away from the component. A component can implement `IDisposable` in a C# base class that implements `BlazorComponent` (or `IComponent` directly). The following component uses `@implements IDisposable` and the `Dispose` method:
```

Comment:
It's not the router that does this. It happens for *all* components when they get removed from the UI (e.g., if they were inside a `if (someBoolean) { ... }` block and `someBoolean` became `false` since the previous render), whether or not that removal was triggered by routing.